kedro-datasets/kedro_datasets_experimental/langfuse/langfuse_prompt_dataset.py

```diff
@@ -0,0 +1,714 @@
+        has_load_args = (self._load_args.get("label") is not None or
+                           self._load_args.get("version") is not None)
+
+        if self._sync_policy == "local" and has_load_args:
+            # Warn user that load_args are ignored in local mode
+            specified_args = []
+            if self._load_args.get("label") is not None:
+                specified_args.append(f"label='{self._load_args['label']}'")
+            if self._load_args.get("version") is not None:
+                specified_args.append(f"version={self._load_args['version']}")
+
+            logger.warning(
+                f"Ignoring load_args ({', '.join(specified_args)}) for prompt '{self._prompt_name}' "
+                f"because sync_policy='local'. Local files are the source of truth. "
+                f"'latest' label is used to fetch remote prompt for synchronization. "
+                f"To use specific versions/labels, switch to sync_policy='remote'."
+            )
+            # Always fetch latest for local sync policy
+            get_kwargs["label"] = "latest"
+        elif self._load_args.get("label") is not None:
+            get_kwargs["label"] = self._load_args["label"]
+        elif self._load_args.get("version") is not None:
+            get_kwargs["version"] = self._load_args["version"]
+        else:
+            get_kwargs["label"] = "latest"
+
+        self._cached_build_args = get_kwargs
+        return self._cached_build_args
+
+    def _sync_strict_policy(
+        self, local_data: str | list | None, langfuse_prompt: Any | None
+    ) -> Any:
+        """Handle strict sync policy - error if local and remote differ.
+
+        Args:
+            local_data: Content from local file, None if file doesn't exist.
+            langfuse_prompt: Langfuse prompt object, None if not found remotely.
+
+        Returns:
+            Langfuse prompt object if sync is successful.
+
+        Raises:
+            DatasetError: If either local_data or langfuse_prompt is missing, or if they differ.
+        """
+        if not local_data or not langfuse_prompt:
+            missing_parts = []
+            if not local_data:
+                missing_parts.append("local file")
+            if not langfuse_prompt:
+                missing_parts.append("remote prompt")
+
+            raise DatasetError(
+                f"Strict sync policy specified for {self._get_prompt_description()}. "
+                f"Both local and remote prompts must exist in strict mode. "
+                f"Missing: {' and '.join(missing_parts)}."
+            )
+
+        local_hash = _hash(_get_content(local_data))
+        remote_hash = _hash(_get_content(langfuse_prompt.prompt))
+        if local_hash != remote_hash:
+            raise DatasetError(
+                f"Strict sync failed for {self._get_prompt_description()}: "
+                f"local and remote prompts differ. Use 'local' or 'remote' policy to resolve."
```

Comment:
When testing the dataset I realised this error doesn't mention langfuse, which might actually help someone debugging. I'm also wondering if we should mention the dataset name here. 
```suggestion
                f"Strict sync to Langfuse failed for {self._get_prompt_description()}: "
                f"local and remote prompts differ. Use 'local' or 'remote' policy to resolve."
```

Reply:
I was thinking the sync policy is for prompts and prompt management so the messages are directed to prompt description. Sync policies are for both local and Langfuse. In case of strict, the policy is to have prompt content same in both local machine and remote (Langfuse). 

If the message says `Strict sync to Langfuse`, for me it feels like we are trying to push content to Langfuse and it failed. My earlier message tries to say the `strict sync failed for a prompt`. Same is the case for  the comment [here](https://github.com/kedro-org/kedro-plugins/pull/1211/files#r2439325120). Let me know if its not clear.

On the dataset name, I wonder how can we get the dataset name within the dataset :(.... also it would be hard if someone is using dataset factories I believe.